apps/core/components/passengerForm/PassengerForm.js

```diff
@@ -48,23 +50,38 @@ const nationalityData = [
   },
 ];
 
+const initialFormState = {
+  gender: 'male',
+  name: null,
+  lastName: null,
+  id: null,
+  nationality: null,
+  dateOfBirth: null,
+  bags: null,
+  passengerCount: 1,
+};
+
 type State = {
   ...PassengerCardType,
   lastName: ?string,
+  hasPrefilled: boolean,
 };
 
 class PassengerForm extends React.Component<Props, State> {
   state = {
-    gender: 'male',
-    name: null,
-    lastName: null,
-    id: null,
-    nationality: null,
-    dateOfBirth: null,
-    bags: null,
-    passengerCount: 1,
+    ...initialFormState,
+    hasPrefilled: false,
   };
 
+  static getDerivedStateFromProps(props, state) {
+    if (!props.isVisible) {
+      return { ...initialFormState, hasPrefilled: false };
+    }
+    return props.isEditing && !state.hasPrefilled
```

Comment:
Also here please add some explanation comment
something like
```
 // If is in edit mode load passenger data into the form state

```

TBH: It took me some time to understand what's going there, so I believe that there exist some more elegant solution.